pkg/metrics/metrics.go

```diff
@@ -194,6 +194,14 @@ func isEndpointsValid(nodes *v1.NodeList, endpoints *v1.Endpoints) bool {
 	for _, node := range nodes.Items {
 		nodeFound := false
 		for _, address := range endpoints.Subsets[0].Addresses {
+			if address.TargetRef == nil {
```

Comment:
I'm curious why we use the `TargetRef` to check the node name rather than the [`NodeName` field](https://pkg.go.dev/k8s.io/api/core/v1#EndpointAddress.NodeName)?

Reply:
Not 100% sure but probably that can be nil in some cases, could be verified and used if that's not the case.

Reply:
Do we know why we ended up in a scenario where the `TargetRef` was nil?

>I'm curious why we use the TargetRef to check the node name rather than the [NodeName field](https://pkg.go.dev/k8s.io/api/core/v1#EndpointAddress.NodeName)?

I have the same question. I would think if `TargetRef` is nil, `NodeName` also would be `nil` but we should confirm that. Either way we need to have `nil` checks.

Reply:
> Do we know why we ended up in a scenario where the TargetRef was nil?

IDK.

Below the object extracted from the case, where only the `ip` is present in the list of addresses.

```yaml
---
apiVersion: v1
kind: EndpointsList
items:
- apiVersion: v1
  kind: Endpoints
  metadata:
    creationTimestamp: "2022-10-27T06:53:02Z"
    labels:
      name: windows-exporter
    name: windows-exporter
    namespace: openshift-windows-machine-config-operator
    resourceVersion: "136231886"
    uid: e4a3e21d-cdcb-4de9-907d-5a9e908aa27c
  subsets:
  - addresses:
    - ip: 10.0.38.87
    - ip: 10.0.38.88
    - ip: 10.0.38.85
    - ip: 10.0.38.84
    - ip: 10.0.38.86
    ports:
    - name: metrics
      port: 9182
      protocol: TCP

```

Reply:
> I would think if TargetRef is nil, NodeName also would be nil but we should confirm that.

In the [docs](https://docs.openshift.com/container-platform/4.11/rest_api/network_apis/endpoints-v1.html#subsets-addresses-2), looks like `TargetRef`  and `NodeName` are independent. 

`NodeName` is [not in use ](https://github.com/openshift/windows-machine-config-operator/blob/c8f92e7244d5a41af51302e715a846efdd7468d3/pkg/metrics/metrics.go#L173) by WMCO as it is set to `nil`.  Hence, only `TargetRef` requires validation.

Reply:
>Below the object extracted from the case, where only the ip is present in the list of addresses.

Please investigate the controller that manages `EndpointList` and ensure this is not pointing at bug in another component. This can be done independently of fixing this bug. We need to figure why the TargetRef` is being removed.

>NodeName is [not in use ](https://github.com/openshift/windows-machine-config-operator/blob/c8f92e7244d5a41af51302e715a846efdd7468d3/pkg/metrics/metrics.go#L173)by WMCO as it is set to nil. Hence, only TargetRef requires validation.

Yes I know that and my question if we should be setting that. PS: if yes, then it needs to be done independently in a story under the monitoring epic.